### PR TITLE
tools: refine pd-heartbeat-bench

### PR DESCRIPTION
**Pull request by HunDunDM**
Signed-off-by: HunDunDM <hundundm@gmail.com>

### What problem does this PR solve?
Issue Number: Close #4399

### What is changed and how it works?

* refine `pd-heartbeat-bench` to bench more complicated situations

### Check List

Tests

- No code
  - Only the code of the bench tool has been modified

### Release note

```release-note
None
```

### CODE REVIEW

File: tools/pd-heartbeat-bench/main.go [hunk 1]
```diff
@@ -193,30 +310,35 @@ func main() {
...
-			startNotifier <- report
+			startNotifier <- repo
 		}
 		// All stores finished heartbeat once.
 		for storeID := 1; storeID <= *storeCount; storeID++ {
 			<-groupEndNotify[storeID]
 		}
-
-		close(report.Results())
+		endTime := time.Now()
```
Comment (JmPotato) [inline tools/pd-heartbeat-bench/main.go:338]: Better use `time.Since` which is more clear.

**Review by JmPotato — COMMENTED** [review]
Rest LGTM.

**Review by nolouch — COMMENTED** [review]
lgtm

File: tools/pd-heartbeat-bench/main.go [hunk 2]
```diff
@@ -126,61 +136,168 @@ func newEndKey(id uint64) []byte {
...
+	// update space
+	for _, i := range rs.updateSpace {
+		region := rs.regions[i]
+		region.ApproximateSize += bytesUnit
+		region.ApproximateKeys += keysUint
+	}
+	// update flow
+	for _, i := range rs.updateFlow {
+		region := rs.regions[i]
+		region.BytesWritten += bytesUnit
```
Comment (nolouch) [inline tools/pd-heartbeat-bench/main.go:231]: maybe we need a param to control the flow change range.

**Review by nolouch — APPROVED** [review]

**Comment by nolouch** [comment]
Can we merge it?

**Pull request merged**